core/modules/cart/store/actions.ts

```diff
@@ -703,38 +699,48 @@ const actions: ActionTree<CartState, RootState> = {
               })
               diffLog.serverResponses.push({ 'status': res.resultCode, 'sku': serverItem.sku, 'result': res })
             } else {
-              clientCartAddItems.push(
-                new Promise(resolve => {
-                  productActionOptions(serverItem).then((actionOtions) => {
-                    dispatch('product/single', { options: actionOtions, assignDefaultVariant: true, setCurrentProduct: false, selectDefaultVariant: false }, { root: true }).then((product) => {
-                      resolve({ product: product, serverItem: serverItem })
-                    })
-                  })
-                })
-              )
+              const getServerCartItem = new Promise(async resolve => {
```

Comment:
Why are you combining Promises with async/await? Please use just one convention (Currently we prefer async/await). So your solution should looks like this:

```
const getServerCartItem = async () => {
 try {
  const actionOptions = await productActionOptions(serverItem)
 
  if (!actionOptions) {
   return null
  }

  const product = await dispatch('product/single', { options: actionOtions, assignDefaultVariant: true, setCurrentProduct: false, selectDefaultVariant: false }, { root: true })
 
 return { product: product, serverItem: serverItem }
  } catch (err) {
   return null
 }
}
```

Reply:
yeah 😅 I should make all as async/await. Only thing that I have to do then is map those functions in Promise.all
`await Promise.all(clientCartAddItems.map(c => c())`

Reply:
done